### PR TITLE
Build: support build single file

### PR DESCRIPTION
**Pull request by fisker**
## Description

`yarn build --file=parser-babel.js`

## Checklist

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [ ] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: scripts/build/build.mjs [hunk 1]
```diff
@@ -150,34 +150,47 @@ async function preparePackage() {
 }
 
 async function run(params) {
-  await execa("rm", ["-rf", "dist"]);
-  await execa("mkdir", ["-p", "dist"]);
-  if (!params.playground) {
-    await execa("mkdir", ["-p", "dist/esm"]);
+  const shouldUseCache = !params.file && !params["purge-cache"];
+  const shouldBuildMetaFiles = !params.playground && !params.file;
```
Comment (sosukesuzuki) [inline scripts/build/build.mjs:154]: This is actually only used as condition for if we should do `preparePackage`. So what about rename to `shouldPreparePackage`.

**Review by sosukesuzuki — COMMENTED** [review]
Just a question, when do you use this feature?

**Comment by sosukesuzuki** [comment]
Also for some reason, our CI fails...

**Comment by fisker** [comment]
> when do you use this feature?

Debug build problem, for example, when parser-postcss.js is broken, I have to disable many lines to only build that file, because I need other files to run test.

File: scripts/build/build.mjs [hunk 2]
```diff
@@ -150,34 +150,47 @@ async function preparePackage() {
...
+  const shouldPreparePackage = !params.playground && !params.file;
+  let configs = bundleConfigs;
+  if (params.file) {
+    configs = configs.filter(({ output }) => output === params.file);
+  } else {
+    await execa("rm", ["-rf", "dist"]);
   }
 
+  await execa("mkdir", ["-p", "dist"]);
+  await execa("mkdir", ["-p", "dist/esm"]);
```
Comment (sosukesuzuki) [inline scripts/build/build.mjs:163]: we should not make `dist/esm` for playground?
```suggestion
  if (!params.playground) {
    await execa("mkdir", ["-p", "dist/esm"]);
  }
```

Comment (fisker) [reply scripts/build/build.mjs:163]: I don't think it matters to make an empty dir.

**Review by sosukesuzuki — APPROVED** [review]

**Pull request merged**